File: src/main/java/com/camackenzie/exvi/core/model/ExerciseEquipment.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.camackenzie.exvi.core.model;

/**
 *
 * @author callum
 */
public class ExerciseEquipment {

    private final String name;

    public ExerciseEquipment(String name) {
        this.name = name;
    }

    public String getName() {
        return this.name;
    }

    @Override
    public boolean equals(Object other) {
        if (other instanceof ExerciseEquipment) {
            return this.name.equalsIgnoreCase(((ExerciseEquipment) other).name);
        }
        return false;
    }
}

File: common/src/commonMain/kotlin/com/camackenzie/exvi/core/model/ExerciseEquipment.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.camackenzie.exvi.core.model

/**
 *
 * @author callum
 */
@kotlinx.serialization.Serializable
data class ExerciseEquipment(val name: String) {

    override fun equals(other: Any?): Boolean {
        return if (other is ExerciseEquipment) {
            name.equals(other.name, ignoreCase = true)
        } else false
    }
}